src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -108,63 +108,11 @@ internal DeltaSet<TStructuralType> CopyChangedValues(IODataAPIHandler apiHandler
-                    //            "Id":1,
-                    //            "Name":"Customer1",
-                    //            "Orders@odata.delta":[
-                    //                {
-                    //                    "@odata.id":"Customers(3)/Orders(1005)",
-                    //                    "Quantity": 1005
-                    //                },
-                    //                {
-                    //                    "Id": 2000,
-                    //                    "Price":200,
-                    //                    "Quantity":90
-                    //                }
-                    //            ]
-                    //        }
-                    //    ]
-                    //}
-
-                    // If we have a request payload above and we are handling the changed values in Orders,
-                    // The apiHandlerOfT is OrdersAPIHandler with parent as Customers(1).
-                    // The object with "@odata.id" will have OrdersAPIHandler with parent as Customers(3).
-                    // The object with id 2000 will have OrdersAPIHandler similar to apiHandlerOfT.
-                    // The codebelow ensures we use the correct handler.
-                    // NOTE: Handler when we have odata.id will always be different than the parent handler.
-                    // However this is true if handlers are applied correctly.
-
-                    ODataAPIHandler<TStructuralType> odataPathApiHandler = apiHandlerFactory.GetHandler(changedObj.ODataPath) as ODataAPIHandler<TStructuralType>;
-
-                    if (odataPathApiHandler != null && changedObj.ODataPath.Any())
-                    {
-                        // Using a 3rd party library to compare objects
-                        // https://github.com/GregFinzer/Compare-Net-Objects/wiki/Getting-Started
-                        // TODO:
-                        // a) Confirm to perf regressions are caused byt his library.
-                        // b) Investigate if there is a better library.
-
-                        KellermanSoftware.CompareNetObjects.CompareLogic compare = new KellermanSoftware.CompareNetObjects.CompareLogic();
-                        compare.Config.CaseSensitive = false;
-                        compare.Config.ComparePrivateProperties = true;
-                        compare.Config.ComparePrivateFields = true;
-
-                        KellermanSoftware.CompareNetObjects.ComparisonResult result = compare.Compare(handler, odataPathApiHandler);
-
-                        if (!result.AreEqual)
-                        {
-                            handler = odataPathApiHandler;
-                            hasODataId = true;
-                        }
-                    }
+                    odataPathApiHandler = apiHandlerFactory.GetHandler(changedObj.ODataPath) as ODataAPIHandler<TStructuralType>;
```

Comment:
I don't think this logic should be here because it means that even when users pass in the handler to use to update resources with,, we still need to execute this logic.. Maybe you move it to where you are substituting the `handler` with the `odataPathApiHandler`

Reply:
Fixed